src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,352 @@
+                    throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+                }
+                int ret = Interop.Stc.Rule.SetIfaceType(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set Interface type, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// A property for Time period for statistics rule.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <value>Time period.</value>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public StcTimePeriod TimePeriod 
+        {
+            get
+            {
+                StcTimePeriod timePeriod ;
+                int ret = Interop.Stc.Rule.GetTimePeriod(_ruleHandle, out timePeriod);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get Time period, Error - " + (StcError)ret);
+                }
+                return timePeriod;
+            }
+            set
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+                }
+                int ret = Interop.Stc.Rule.SetTimePeriod(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set Time period, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information an application matched rule asynchronously.
+        /// </summary>
+        /// <since_tizen> tizen_5.5 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <param name="userData">The user data passed to the callback function.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetStats(StatsInfoObjectCb.InfoCallback infoCb, IntPtr userData)
```

Comment:
Why don't you provide async method?
```suggestion
        public Task<StcInfo> GetStats()
```

In C# userData never needed.
because C# provide a closure concept

Reply:
1) userData is removed from function parameter as per your suggestion.
2) The callback is called for each application to provide its information. So i can't  find what we have to await for inside this method. As we don't know how many times this callback will be called.

Reply:
```suggestion
        public IEnumerable<StcInfo> GetStats();
```

Reply:
Please study about C# async / await, Task